AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -516,10 +503,18 @@ - (void)invalidateDataSourceItemCounts
 
 - (void)beginUpdates
 {
-  dispatch_group_wait(_editingTransactionGroup, DISPATCH_TIME_FOREVER);
-  // Begin queuing up edit calls that happen on the main thread.
-  // This will prevent further operations from being scheduled on _editingTransactionQueue.
-  _batchUpdateCounter++;
+  ASDisplayNodeAssertMainThread();
+  // TODO: -waitUntilAllUpdatesAreCommitted
```

Comment:
I restored the previous behavior of dispatch_group_wait for now. I haven't completely thought through promoting that to -waitUntilAllUpdatesAreCommitted. Doing it right now causes some internal issues so I just added the note that we should maybe do it later. No behavior change around this for the moment.